tests/commands/new/test_new_app.py

```diff
@@ -46,18 +53,79 @@ def test_new_app(monkeypatch, new_command, tmp_path, briefcase_version):
     # Template is updated
     new_command.update_cookiecutter_cache.assert_called_with(
         template="https://github.com/beeware/briefcase-template",
-        branch="v37.42.7",
+        branch=expected_branch,
     )
     # Cookiecutter is invoked
     new_command.cookiecutter.assert_called_with(
         "~/.cookiecutters/briefcase-template",
         no_input=True,
         output_dir=os.fsdecode(tmp_path),
-        checkout="v37.42.7",
+        checkout=expected_branch,
         extra_context=app_context,
     )
 
 
+def test_new_app_dev(monkeypatch, new_command, tmp_path):
+    """In a dev version, template will fall back to the 'main' branch if a
+    versioned template doesn't exist."""
```

Comment:
There should also be a test that it *doesn't* fall back to the main branch in a non-dev version, but throws an error. And likewise if the user explicitly requests a branch that doesn't exist.